processors/tenantidprocessor/tenantidprocessor.go

```diff
@@ -63,3 +90,58 @@ func (p *processor) addTenantIdToSpans(traces pdata.Traces, tenantIDHeaderValue
 		}
 	}
 }
+
+func (p *processor) addTenantIdToMetrics(metrics pdata.Metrics, tenantIDHeaderValue string) {
+	rms := metrics.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		rm := rms.At(i)
+		ilms := rm.InstrumentationLibraryMetrics()
+		for j := 0; j < ilms.Len(); j++ {
+			ilm := ilms.At(j)
+			metrics := ilm.Metrics()
+			for k := 0; k < metrics.Len(); k++ {
+				metric := metrics.At(k)
+				metricDataType := metric.DataType().String()
+				switch metricDataType {
+				case "None":
+					p.logger.Error("Cannot add tenantId to metric. Metric Data type not present for metric: " + metric.Name())
```

Comment:
@davexroth Should we just log it and allow the metrics to move forward without tenantId or should we return error and drop metrics?

Reply:
Never mind, prometheus exporter drops them anyways if the DataType is none